packages/fast-usdc-contract/src/exos/advancer.ts

```diff
@@ -278,24 +313,67 @@ export const prepareAdvancerKit = (
           result: undefined,
           ctx: AdvancerVowCtx & { tmpSeat: ZCFSeat },
         ) {
-          const { poolAccount, intermediateRecipient, settlementAddress } =
-            this.state;
-          const { destination, advanceAmount, tmpSeat, ...detail } = ctx;
-          tmpSeat.exit();
-          const amount = harden({
-            denom: usdc.denom,
-            value: advanceAmount.value,
-          });
-          const transferOrSendV =
-            destination.chainId === settlementAddress.chainId
-              ? E(poolAccount).send(destination, amount)
-              : E(poolAccount).transfer(destination, amount, {
-                  forwardOpts: { intermediateRecipient },
-                });
-          return watch(transferOrSendV, this.facets.transferHandler, {
-            destination,
-            advanceAmount,
-            ...detail,
+          return asVow(async () => {
```

Comment:
We typically use `asVow` when we intend to return a vow and something might throw synchronously early. `asVow` will ensure that's wrapped up a vow rejection.

Here, it doesn't do much as no one is waiting on the result of `advancer.handleTransactionEvent`. We need to handle all errors here, log loudly if necessary, and ensure the proper state transition still occurs.

Reply:
> Am I right to infer we should call statusManager.skipAdvance()

If we can do precondition checks before `statusManager.advance(evidence);` on ~L220, yes. If we already started attempting we should use `notifier.notifyAdvancingResult(evidence, false)` and perform `borrower.returnToPool`